src/Sulu/Bundle/CategoryBundle/Entity/Category.php

```diff
@@ -334,6 +334,22 @@ public function getTranslations()
     }
 
     /**
+     * Get single meta by locale.
+     *
+     * @param $locale
+     *
+     * @return CategoryTranslation
+     */
+    public function findTranslationByLocale($locale)
+    {
+        return $this->translations->filter(
+            function (CategoryTranslation $translation) use ($locale) {
+                return $translation->getLocale() === $locale;
```

Comment:
this is quite performant because all translations will be automatically loaded and it don't need another sql query.